builder/vsphere/iso/step_create.go

```diff
@@ -105,10 +103,6 @@ func (c *CreateConfig) Prepare() []error {
 		c.GuestOSType = "otherGuest"
 	}
 
-	if c.Firmware != "" && c.Firmware != "bios" && c.Firmware != "efi" && c.Firmware != "efi-secure" {
-		errs = append(errs, fmt.Errorf("'firmware' must be 'bios', 'efi' or 'efi-secure'"))
-	}
-
```

Comment:
Validating this allows to fail a bit faster and also to fail in the case we run `packer validate` .